src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -213,6 +213,37 @@ object AST extends Positional {
         null
     }
   }
+
+  def evalComposeNumericArray[T](ec: EvalContext, subexpr: AST)(g: (IndexedSeq[T]) => Any)
```

Comment:
yeah, this is slightly better.  But I'd really like to mix and match arrays and scalars, and even if there's an evalCompose for arrays, that doesn't solve the problem.